src/main/java/io/sniffy/util/StackTraceExtractor.java

```diff
@@ -37,6 +37,34 @@ private StackTraceExtractor() {
         }
     }
 
+    // TODO: refactor this method
+    public static List<StackTraceElement> getTraceTillPackage(String packageName) {
+        StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+        // skip all elements until proxied call
+        int startIndex = 0;
+        for (int i = 1; i < stackTraceElements.length; i++) {
+            StackTraceElement traceElement = stackTraceElements[i];
+            String traceElementClassName = traceElement.getClassName();
+            if (!traceElementClassName.startsWith(packageName) &&
+                    !traceElementClassName.startsWith("java.security") &&
+                    !traceElementClassName.startsWith("com.sun") &&
+                    !traceElementClassName.equals("io.sniffy.socket.SnifferSocketImpl") &&
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/sniffy/sniffy/file/2495411458/issues/source?bid=3005960&fileBranchId=3034483#l51)